setup_configure.py

```diff
@@ -58,6 +58,7 @@ class EnvironmentOptions(object):
     def __init__(self):
         self.hdf5 = os.environ.get('HDF5_DIR')
         self.hdf5_version = os.environ.get('HDF5_VERSION')
+        self.mpi = os.environ.get('HDF5_MPI') == "ON"
```

Comment:
For mpl we use every value as `True`, and the env not being defined as `False`, but we may not be following conventions well enough.

For simplicity I agree just one 'true' value is better than a mid-sized set.